peps/pep-0776.rst

```diff
@@ -0,0 +1,792 @@
+specified here:
+
+* https://webassembly.github.io/spec/core/binary/index.html
+* https://github.com/WebAssembly/tool-conventions/blob/main/DynamicLinking.md
+
+The ``emcc`` compiler is a wrapper around ``clang``. The ``emcc`` linker is a
+wrapper around ``wasm-ld`` (also part of the LLVM toolchain).
+
+Emscripten support for portable C/C++ code source compatibility with Linux is
+fairly comprehensive, with certain expected exceptions to be spelled out.
+CPython already supports compilation to Emscripten, and it only requires a very
+modest number of modifications to the normal Linux target.
+
+----------------
+POSIX Compliance
+----------------
+
+Emscripten is a POSIX platform. However, there are POSIX APIs that exist but
+always fail when called and POSIX APIs that don't exist at all. In particular,
+there are problems with networking APIs and blocking I/O, and there is no
+support for ``fork()``. See `Emscripten Portability Guidelines
+<https://emscripten.org/docs/porting/guidelines/portability_guidelines.html>`__.
+
+Emscripten executables can be linked with threading support, however, it comes
+with several limitations:
+
+* Enabling threading requires websites to be served with special security headers
+  that indicate acceptance of the possibility of spectre-style information
+  leakage. These headers are a usability hazard for users who are not intimately
+  familiar with the web platform.
+
+* If an executable is linked with both threading and a dynamic loader, Emscripten
+  prints a warning that using dynamic loading and pthreads together is
+  experimental. It may cause performance problems or crashes. These problems may
+  require WebAssembly standards work to resolve.
+
+Because of these limitations, Pyodide standardizes a no-pthreads build of
+Python. We will start with support for no-pthreads builds. If there is
+sufficient demand, a pthreads build with no dynamic loader could be added later.
+
+----------------------------
+Emscripten ABI Compatibility
+----------------------------
+
+The Emscripten compiler has no ABI stability guarantees between versions. Many
+Emscripten updates are ABI compatible by chance, and the Rust Emscripten target
+behaves as if the ABI were stable with only `occasional negative consequences
+<https://github.com/rust-lang/rust/issues/131467>`__.
+
+There are several linker flags that adjust the Emscripten ABI, so Python
+packages built to run with Emscripten must make sure to match the ABI-sensitive
+linker flags used to compile the interpreter to avoid load-time or run-time
+errors. The Emscripten compiler continuously fixes bugs and adds support for new
+web platform features. Thus, there is significant benefit to being able to
+update the ABI.
+
+In order to balance the ABI stability needs of package maintainers with the ABI
+flexibility to allow the platform to move forward, we plan to adopt a new ABI
+for each minor (3.xx) release of Python.
```

Comment:
We call them feature releases elsewhere: https://devguide.python.org/developer-workflow/development-cycle/

```suggestion
for each feature release of Python.
```

Reply:
thanks!